### ISSUE TITLE
Pretrained Hdemucs model

### ISSUE THREAD

**Issue by shyama-20** [facebookresearch/demucs#316]
## ❓ Questions

Hello,
i) Is it possible to retrain the already trained demucs model with our custom data?
ii) Is it possible to get access to any of the 4 Track A models mentioned in the training document?
Thank you!

**Comment by adefossez** [comment]
Yes to both. To initialize from a pre-train model, just pass `continue_pretrained=SIG` where SIG is one of the signature of the individual models. You will also need to use the right hyper params for the model (this is not done automatically). To continue from the model `c511e2ab`, you can do 
```
dora run -f 81de367c -d continue_pretrained=c511e2ab [EXTRA ARGS]
```
I'm using `81de367c` and not `c511e2ab`after the `-f` flag because the latter is the fine tuned one, but you probably want to use the original training parameter, before fine tuning. The `-f` flag means "inject all the hyper params from the given XP". The EXTRA ARGS are passed after the ones from the XP given by `-f`.

You can get any of the models programmatically:
```
from demucs import pretrained
pretrained.get_model("c511e2ab")
```

**Comment by shyama-20** [comment]
@adefossez 

Thank you for your reply.

I got the model downloaded using the program and obtained the `.argv.json` file as shown below

![Screenshot from 2022-04-06 12-29-13](https://user-images.githubusercontent.com/90311325/161920371-fb01dd42-53a5-42e2-ae89-9a13b901fab2.png)

Please correct me if I'm wrong but what I understood is that we should update the config.yaml file alone according to these hyper parameters shown above and then run `dora run -f 81de367c -d continue_pretrained=c511e2ab dset=my_dset`to retrain with our custom data (path is provided in `my_dset.yaml`)

Following this approach, I got the this Runtime error

RuntimeError: Error(s) in loading state_dict for HDemucs:
	size mismatch for decoder.5.conv_tr.weight: copying a param with shape torch.Size([48, 8, 8, 1]) from checkpoint, the shape in current model is torch.Size([48, 4, 8, 1]).
	size mismatch for decoder.5.conv_tr.bias: copying a param with shape torch.Size([8]) from checkpoint, the shape in current model is torch.Size([4]).
	size mismatch for tdecoder.4.conv_tr.weight: copying a param with shape torch.Size([48, 8, 8]) from checkpoint, the shape in current model is torch.Size([48, 4, 8]).
	size mismatch for tdecoder.4.conv_tr.bias: copying a param with shape torch.Size([8]) from checkpoint, the shape in current model is torch.Size([4]).

Can you please point out why this mismatch is occurring?

**Comment by FSharpCSharp** [comment]
> Yes to both. To initialize from a pre-train model, just pass `continue_pretrained=SIG` where SIG is one of the signature of the individual models. You will also need to use the right hyper params for the model (this is not done automatically). To continue from the model `c511e2ab`, you can do
> 
> ```
> dora run -f 81de367c -d continue_pretrained=c511e2ab [EXTRA ARGS]
> ```
> 
> I'm using `81de367c` and not `c511e2ab`after the `-f` flag because the latter is the fine tuned one, but you probably want to use the original training parameter, before fine tuning. The `-f` flag means "inject all the hyper params from the given XP". The EXTRA ARGS are passed after the ones from the XP given by `-f`.
> 
> You can get any of the models programmatically:
> 
> ```
> from demucs import pretrained
> pretrained.get_model("c511e2ab")
> ```

I also have a question that goes in a similar direction. What are the recommended hardware requirements for training? With the original demucs variant, these were quite high at the time (you couldn't have enough RAM). Unfortunately, I can't find any information about this on the training description page. Therefore, it would be great if the minimum hardware requirements could be added. Especially with regard to graphics cards, this would be very interesting to know.

**Comment by adefossez** [comment]
For the original training parameters, 32GB of memory are required (maybe 24GB would work but I'm not sure). I used 8 GPUs, so with a single GPU you would need a batch size of 8. With a batch size of 4, you can train with 16GB.
If you have less memory, then you will need to reduce either the training segment duration, or the number of channels in the model (see https://github.com/facebookresearch/demucs/issues/313 for how to do this). You can also further reduce the batch size, although a batch size of 1 will break the shuffling of the source.

**Comment by FSharpCSharp** [comment]
Thank you for the quick and competent response @adefossez! Since the topic unfortunately seems to be distributed over several issues, it would be great for the future if you could note all this information and notes on the training.md file. Then many questions would be clarified from the outset! Just as a small suggestion.

**Comment by KimberleyJensen** [comment]
@shyama-20 I think you are getting this error because the model you are trying to train from has 4 sources but your dset (https://github.com/facebookresearch/demucs/issues/313#issue-1189561074) has 2
